src/clients/python/src/tigerbeetle_python/lib.py

```diff
@@ -0,0 +1,68 @@
+import ctypes
+import platform
+from pathlib import Path
+
+
+class NativeError(Exception):
+    pass
+
+class IntegerOverflowError(Exception):
+    pass
+
+
+def _load_tbclient():
+    prefix = ""
+    arch = ""
+    system = ""
+    linux_libc = ""
+    suffix = ""
+
+    if platform.machine() == "x86_64" or platform.machine() == "AMD64":
+        arch = "x86_64"
+    elif platform.machine() == "aarch64":
+        arch = "aarch64"
+    else:
+        raise NativeError("Unsupported machine: " + platform.machine())
+
+    if platform.system() == "Linux":
+        prefix = "lib"
+        system = "linux"
+        suffix = ".so"
+        libc = platform.libc_ver()[0]
+        if libc == "glibc":
+            linux_libc = "-gnu.2.27"
+        elif libc == "musl":
+            linux_libc = "-musl"
+        else:
+            raise NativeError("Unsupported libc: " + libc)
+    elif platform.system() == "Darwin":
+        prefix = "lib"
+        system = "macos"
+        suffix = ".dylib"
+    elif platform.system() == "Windows":
+        system = "windows"
+        suffix = ".dll"
+    else:
+        raise NativeError("Unsupported system: " + platform.system())
+
+    source_path = Path(__file__).resolve()
```

Comment:
Would it work without `.resovle()`? The rule of thumb for paths is to avoid canonicalization, and use them as is.